docs/scaling.md

```diff
@@ -0,0 +1,50 @@
+# Scaling the number of Firecracker microVMs per host
+
+To scale the number of microVMs past one thousand, one needs to properly configure 
+the system constraints and provision enough networking resources for the microVMs.
+
+On Ubuntu 18.04 Linux, one needs to properly configure the number of processes, threads, memory, 
+open files that may simultaneously exist in a system as well as create enough 
+virtual bridges (one bridge can serve up to 1023 interfaces).
+
+To configure the system, one needs to set up the following parameters. 
+Note that the exact values depend on your setting.
+
+In `/etc/security/limits.conf`, set `nofile`, `nproc` and `stack` to the 
+appropriate values for both normal users and root:
+```
+sudo sh -c "echo \"* soft nofile 1000000\" >> /etc/security/limits.conf"
+sudo sh -c "echo \"* hard nofile 1000000\" >> /etc/security/limits.conf"
+sudo sh -c "echo \"root soft nofile 1000000\" >> /etc/security/limits.conf"
+sudo sh -c "echo \"root hard nofile 1000000\" >> /etc/security/limits.conf"
+sudo sh -c "echo \"* soft nproc 4000000\" >> /etc/security/limits.conf"
+sudo sh -c "echo \"* hard nproc 4000000\" >> /etc/security/limits.conf"
+sudo sh -c "echo \"root soft nproc 4000000\" >> /etc/security/limits.conf"
+sudo sh -c "echo \"root hard nproc 4000000\" >> /etc/security/limits.conf"
+sudo sh -c "echo \"* soft stack 65536\" >> /etc/security/limits.conf"
+sudo sh -c "echo \"* hard stack 65536\" >> /etc/security/limits.conf"
+sudo sh -c "echo \"root soft stack 65536\" >> /etc/security/limits.conf"
+sudo sh -c "echo \"root hard stack 65536\" >> /etc/security/limits.conf"
```

Comment:
For this section, it would make sense to show that as a file, rather then the sequence of `echo` commands like https://github.com/firecracker-microvm/firecracker-containerd/blob/master/docs/getting-started.md#configure-firecracker-containerd-binary.

Regarding the configuration itself, I have a few questions.

- Do we need to have `root` and `*` separately? Doesn't `*` cover `root`?
- Assuming that Firecracker is running as a separate Linux user, not root, can we specifically target the user instead of essentially removing the limits for all of them?

Reply:
@kzys Thank you for the feedback, I have applied it. 

- root and the rest of the users are managed separately in Ubuntu, please see [here](https://stackoverflow.com/questions/38991351/why-in-etc-security-limits-conf-doesnt-include-root-user) for details. 
- The orchestrator (like the one in `examples/taskworkflow.go`) runs with sudo (at least in the quickstart/getting-started example commands) and it creates a lot of threads and processes As a result, it can hit one of the system limits as well as VMs themselves.

Reply:
Thanks! I didn't know that `*` doesn't include `root`. 

I can understand why we need more `nofile` and `nproc`, but not so sure about `stack`. Does something consume more stack by having more VMs?